code/actions/UnpublishItemWorkflowAction.php

```diff
@@ -0,0 +1,76 @@
+<?php
+/**
+ * Unpublishes an item
+ *
+ * @author     marcus@silverstripe.com.au
+ * @license    BSD License (http://silverstripe.org/bsd-license/)
+ * @package    advancedworkflow
+ * @subpackage actions
+ */
+class UnpublishItemWorkflowAction extends WorkflowAction {
+
+	private static $db = array(
+		'PublishDelay' => 'Int'
+	);
+
+	public static $icon = 'advancedworkflow/images/publish.png';
+
+	public function execute(WorkflowInstance $workflow) {
+		if (!$target = $workflow->getTarget()) {
+			return true;
+		}
+
+		if (class_exists('AbstractQueuedJob') && $this->PublishDelay) {
+			$job   = new WorkflowPublishTargetJob($target);
+			$days  = $this->PublishDelay;
+			$after = date('Y-m-d H:i:s', strtotime("+$days days"));
+			singleton('QueuedJobService')->queueJob($job, $after);
+		} else if ($target->hasExtension('WorkflowEmbargoExpiryExtension')) {
+			// setting future date stuff if needbe
+
+			// set these values regardless
+			$target->DesiredUnPublishDate = '';
+			$target->DesiredPublishDate = '';
+			$target->write();
+			
+			if ($target->hasMethod('doUnpublish')) {
+				$target->doUnpublish();
+			}
+		} else {
+			if ($target->hasMethod('doUnpublish')) {
+				$target->doUnpublish();
+			}
+		}
+
+		return true;
+	}
+
+	public function getCMSFields() {
+		$fields = parent::getCMSFields();
+
+		if (class_exists('AbstractQueuedJob')) {
+			$before = _t('UnpublishItemWorkflowAction.DELAYUNPUBDAYSBEFORE', 'Delay publication ');
+			$after  = _t('UnpublishItemWorkflowAction.DELAYUNPUBDAYSAFTER', ' days');
+
+			$fields->addFieldToTab('Root.Main', new FieldGroup(
+				_t('UnpublishItemWorkflowAction.UNPUBLICATIONDELAY', 'Publication Delay'),
+				new LabelField('UnpublishDelayBefore', $before),
+				new NumericField('UnpublishDelay', ''),
+				new LabelField('UnpublishDelayAfter', $after)
+			));
+		}
+
+		return $fields;
+	}
+
+	/**
+	 * Unpublish action allows a user who is currently assigned at this point of the workflow to
+	 *
+	 * @param  DataObject $target
+	 * @return bool
+	 */
+	public function canUnpublishTarget(DataObject $target) {
```

Comment:
This probably should be "canPublishTarget" - canUnpublishTarget isn't used anywhere.